db/commitlog/commitlog.cc

```diff
@@ -1696,7 +1708,7 @@ void db::commitlog::segment_manager::on_timer() {
             if (max != 0 && cur >= max) {
                 _new_counter = 0;
                 clogger.debug("Used size on disk {} MB exceeds local maximum {} MB", cur / (1024 * 1024), max / (1024 * 1024));
-                flush_segments();
+                flush_segments(cur - max/2);
```

Comment:
Why not cur - max? since max is the threshold.

btw, I don't see why it should be in the timer. As soon as a segment is completed, we can consider it against the threshold. The timer only makes it less predictable.

Reply:
We can add another eager re-time of the background on segment finish, but I think it is better to keep the code paths separate. 
As for the size thingamajing, my reasoning is that requesting only cur-max would make us end up here more often - better to try to get a little larger portion, esp if we find we can tune up the threshold to something like 2/3 of actual max, which might be better, assuming flush can keep up.

Reply:
Ah, hysteresis. But doesn't that take us to 1/4 of the maximum? Since the threshold is around 1/2 of commitlog? I think that is too eager.

Reply:
Maybe. Worth tuning. But can change initial value.